stumpy/core.py

```diff
@@ -2561,6 +2575,174 @@ def _select_P_ABBA_value(P_ABBA, k, custom_func=None):
     return MPdist
 
 
+@njit
+def _merge_topk_PI(PA, PB, IA, IB):
+    """
+    Merge two top-k matrix profiles `PA` and `PB`, and update `PA` (in place) while
+    always prioritizing the values of `PA` over the values of `PB` in case of ties.
+    (i.e., values from `PB` are always inserted to the right of values from `PA`).
+    Also, update `IA` accordingly. In case of overlapping values between two arrays
+    IA[i] and IB[i], the ones in IB[i] (and their corresponding values in PB[i])
+    are ignored throughout the updating process of IA[i] (and PA[i]).
+
+    Unlike `_merge_topk_ρI`, where `top-k` largest values are kept, this function
+    keeps `top-k` smallest values.
+
+    Parameters
+    ----------
+    PA : numpy.ndarray
+        A (top-k) matrix profile where values in each row are sorted in ascending
+        order. `PA` must be 2-dimensional.
+
+    PB : numpy.ndarray
+        A (top-k) matrix profile where values in each row are sorted in ascending
+        order. `PB` must have the same shape as `PA`.
+
+    IA : numpy.ndarray
+        A (top-k) matrix profile indices corresponding to `PA`
+
+    IB : numpy.ndarray
+        A (top-k) matrix profile indices corresponding to `PB`
+
+    Returns
+    -------
+    None
+    """
+    k = PA.shape[1]
+    tmp_P = np.empty(k, dtype=np.float64)
+    tmp_I = np.empty(k, dtype=np.int64)
+    for i in range(PA.shape[0]):
+        overlap = set(IB[i]).intersection(set(IA[i]))
+        aj, bj = 0, 0
+        idx = 0
+        for _ in range(2 * k):  # 2 * k to traverse both A and B
+            if idx >= k:
+                break
+
```

Comment:
Remove blank line

Reply:
Have you added additional unit tests to ensure that both merge functions can successfully merge `IA` and `IB` that have overlaps? This is somewhat different from `stump`/`scrump` functions having overlaps

Reply:
Yes. I mean I think so :) So, instead of two test functions `test_merge_topk_PI` and ` test_merge_topk_ρI`, we now have four test functions:

* test_merge_topk_PI_without_overlap
* test_merge_topk_PI_with_overlap
* test_merge_topk_ρI_without_overlap
* test_merge_topk_ρI_with_overlap

I think what you are looking for are `test_merge_topk_PI_with_overlap`  and `test_merge_topk_ρI_with_overlap` where I created overlaps between IA and IB (and thus PA and PB) and test if they can be successfully merged.